zk_toolbox/crates/zk_supervisor/src/commands/test/all.rs

```diff
@@ -0,0 +1,95 @@
+use std::thread;
+
+use common::{cmd::Cmd, config::global_config, logger};
+use xshell::{cmd, Shell};
+
+use super::{
+    args::{
+        all::AllArgs, integration::IntegrationArgs, recovery::RecoveryArgs, revert::RevertArgs,
+        upgrade::UpgradeArgs,
+    },
+    integration, recovery, revert, upgrade,
+};
+
+pub fn run(shell: &Shell, args: AllArgs) -> anyhow::Result<()> {
+    logger::info("Run integration tests");
+    integration::run(
+        shell,
+        IntegrationArgs {
+            external_node: false,
+            no_deps: args.no_deps,
+        },
+    )?;
+
+    logger::info("Run recovery tests (from snapshot)");
+    recovery::run(
+        shell,
+        RecoveryArgs {
+            snapshot: true,
+            no_deps: args.no_deps,
+        },
+    )?;
+
+    logger::info("Run recovery tests (from genesis)");
+    recovery::run(
+        shell,
+        RecoveryArgs {
+            snapshot: false,
+            no_deps: args.no_deps,
+        },
+    )?;
+
+    logger::info("Run external-node");
+    let _handle = thread::spawn(move || {
+        let chain = global_config().chain_name.clone();
+
+        let server_shell = Shell::new().unwrap();
+        let mut cmd =
+            cmd!(server_shell, "zk_inception external-node run").arg("--ignore-prerequisites");
+
+        if let Some(chain) = chain {
+            cmd = cmd.arg("--chain").arg(chain);
+        }
+
+        let _out = Cmd::new(cmd).run_with_output().unwrap();
+    });
+
+    logger::info("Run integration tests (external node)");
+    integration::run(
+        shell,
+        IntegrationArgs {
+            external_node: true,
+            no_deps: args.no_deps,
+        },
+    )?;
+
+    logger::info("Run revert tests");
+    revert::run(
+        shell,
+        RevertArgs {
+            enable_consensus: false,
+            external_node: false,
+            no_deps: args.no_deps,
+        },
+    )?;
+
+    logger::info("Run revert tests (external node)");
+    revert::run(
+        shell,
+        RevertArgs {
+            enable_consensus: false,
+            external_node: true,
+            no_deps: args.no_deps,
+        },
+    )?;
```

Comment:
do we really need to run it twice? seems it's the same test, but one involved revrting the en and other is not

Reply:
Addressed in https://github.com/matter-labs/zksync-era/pull/2594/commits/bb4a83b567e2ed4efab190bc3365a8cdd1e2eb13